### ISSUE TITLE
Full screen title bar not hidden

### ISSUE THREAD

**Issue by russel** [Me-TV/Me-TV#13]
Follow on from Issue #11 

@tomidis Could you put here the problem you are having with the title bar and full screen mode as you are getting it. This is a "it works for me" situation so I need to find out how to replicate the behaviour you are seeing.

**Comment by tomidis** [comment]
No, not the control bar which is hidden.
Only the title bar is visible.
Title bar have only "Me TV" in left corner.
When moving the mouse, control bar appears.
When not moving the mouse, the control bar hides, but the title bar stays (which is the half height of the whole control bar)

**Comment by russel** [comment]
Hummm… when I put it to full screen, it goes full screen, no title bar at all. I am not sure what is happening that you see the title bar. :-(

It sounds like the control bar is doing the same for both of us.

**Comment by russel** [comment]
I have not been able to recreate here the behaviour you are seeing. This makes it impossible for me to investigate what the cause might be. I am at a loss as to how to proceed to improve your experience of full screen.  :-(

**Comment by tomidis** [comment]
I don't know how to help you with this issue.
I uploaded the screenshots:
[https://imgur.com/a/umNlB6C](https://imgur.com/a/umNlB6C)
I used Shutter for the screenshots, but it's impossible to take a quality screenshot with motion picture.

**Comment by russel** [comment]
Thanks for posting that image, most useful to see what is happening on your system. Can I get you to set out some details of your platform: I am wondering if there are settings to the GTK+ calls that I have  that don't affect my usage but do affect your usage.

I am working on Debian Sid, with GNOME.
```
|> uname -a
Linux anglides 4.18.0-2-amd64 #1 SMP Debian 4.18.10-2 (2018-10-07) x86_64 GNU/Linux

|> dpkg -l gnome-shell
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name           Version      Architecture Description
+++-==============-============-============-=====================================
ii  gnome-shell    3.30.1-2     amd64        graphical shell for the GNOME desktop

|> dpkg -l libgtk-3-dev
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name               Version      Architecture Description
+++-==================-============-============-======================================
ii  libgtk-3-dev:amd64 3.24.1-2     amd64        development files for the GTK+ library
```

**Comment by tomidis** [comment]
|> uname -a
Linux admin 4.15.0-34-generic #37-Ubuntu SMP Mon Aug 27 15:21:48 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

|> dpkg -l gnome-shell
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name           Version      Architecture Description
+++-==============-============-============-=================================
un  gnome-shell    <none>       <none>       (no description available)

|> dpkg -l libgtk-3-dev
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name           Version      Architecture Description
+++-==============-============-============-=================================
ii  libgtk-3-dev:a 3.22.30-1ubu amd64        development files for the GTK+ li


I'm not sure how to use "Insert code" when writing message, to get formatted text like you do?

**Comment by russel** [comment]
I use three backticks above and three backticks below as I do it manually, but I think you can use the <> button above the write area to do it in a more word processors way.

**Comment by russel** [comment]
Does the lack  of GNOME Shell mean you are using Unity, or perhaps one of the other window managers?

**Comment by tomidis** [comment]
Cinnamon v3.8.9

**Comment by russel** [comment]
Aha, I think we now have a hook with which to try and catch the problem. My hypothesis is that the GNOME Shell orientation of the code is not driving GTK+ correctly for use with Cinnamon. I'll hunt around and see if I can find something. But not in the  next couple of days I suspect. Sorry, but I have to prepare for some travelling and other stuff.

**Comment by russel** [comment]
Apologies for the delay in getting back to this – long story.

Is the bar present in windowed mode as well as full screen mode?

I am assuming the problem is that GTK+ features being used in Me TV are behaving differently on Gnome and Cinnamon because that is a difference between Gnome and Cinnamon. "Statin' the bleedin' obvious" true, but the point is I suspect I am never going to be able to reproduce this on Gnome. This means I shall have to find a way of running Cinnamon in a container/virtual machine on Debian.

**Comment by SparksDJ** [comment]
I'm using me-tv on Debian Buster (my wife's new box is too new for Stretch) and Gnome Flashback.  I see an extra bar in fullscreen mode, but it's the menu bar, not the title bar.  The title bar disappears when the mouse and keyboard have been inactive.

In fullscreen mode the menu bar is *above* the title bar; in windowed mode the menu bar is *below* the title bar.

You should be able to select Gnome Flashback when logging in on Debian.

(I had intended to email you to introduce myself, but since you have so few users, I might as well di it here.  I'm UK-based, retired with 50 years programming experience, including mainframe operating systems and compiler run-time libraries.  I've no GUI experience.)

**Comment by russel** [comment]
@SparksDJ  Welcome to the exclusive club of Me TV version 3 users. :-)

Thanks for chipping in. After the Gnome 2 → Gnome 3 revolution and the introduction of GnomeShell, I tried xfce and it didn't work for me. Gnome Fallback didn't really "cut it", so I forced myself to actually try GnomeShell, and discovered it worked very well for me. I therefore ignored Gnome Fallback, and hadn't realised that there was a Gnome Flashback. Clearly the window header and menu behaviour of Gnome Flashback is similar to the Cinnamon behaviour and so offers a place I can try things out easily. I shall install the packages and give it a go logging in. If it shows the problem @tomidis reported but on my machine, it is going to be easier to investigate. It seems fairly clear that it is something to do with the way application menu bars are handled by GnomeShell and not-GnomeShell.

(Sounds like you have been at this programming lark longer than I have! I started in scientific computation, went via parallelism, to HCI and socio-technical systems, and now video and audio with GUIs. Surprisingly even though the context and goals are different the actual programming always ends up with the same architectures and problems.)

**Comment by russel** [comment]
@tomidis @SparksDJ proposal of Gnome Flashback turns out to be a really splendid suggestion. :-)

So the images at https://imgur.com/a/umNlB6C are what happens when the application menu exists but it is not GnomeShell. So this is not so much a bug, more a question of what Me TV should do with the application menu in non-GnomeShell situations.

**Comment by russel** [comment]
Hummm… at least on Gnome Flashback the application menu is present on the application window, which means that being on the television window in either window or full screen mode is actually just wrong. I have answered my own question, and I think know the problem and solution. The question is then why wasn't this a problem on GnomeShell?

**Comment by russel** [comment]
Well that turned out to be easy to fix: the television windows were erroneously application windows rather than toplevel windows. This was not observable with GnomeShell but certainly was with Gnome Flashback. I believe Cinnamon should behave as Gnome Flashback.

I'll close this as fixed, in the hope that it is. I'll make a new bug fix release.

**Comment by tomidis** [comment]
Excellent, you solved this issue until I checked all messages :+1: 
I can confirm, this issue is fixed in v3.0.6

**Comment by russel** [comment]
@tomidis Thanks for checking and reporting back in. 

Thanks to @SparksDJ for providing the solution to the how do I test this locally!
